### DISCUSSION TITLE
Error after Symfony installation with permissions to write to directory in windows 10.

### DISCUSSION THREAD

**Discussion by wnunezc** [symfony/symfony#44301] [Q&A]
### Symfony version(s) affected

5.3.7

### Description

I am starting with symfony and at the end of the installation without having touched anything I get this error:

[![introducir la descripción de la imagen aquí][1]][1]

**my environment:** I am using windows 10 and Laragon (vitaminized xampp)

checking the directory I realize it has this in the attribute:

[![introducir la descripción de la imagen aquí][2]][2]

but when validating the permissions these are the ones marked:

[![introducir la descripción de la imagen aquí][3]][3]

That is the List of Users with permissions and all are equal with read and write permission.

So I don't know what I need ... there is a persistence in the error ...

  [1]: https://i.stack.imgur.com/sDm3Q.png
  [2]: https://i.stack.imgur.com/vTtV1.png
  [3]: https://i.stack.imgur.com/c4lD4.png

### How to reproduce

Follow the steps in the description!

### Possible Solution

I do not understand why this happens I have realized that it does not matter how many times I make changes in the read-only properties or in the permissions of the users; the rea-only property does not change and the permissions of the users despite being placed do not finish working for me ...

to test and validate all of this, I have altered the symfony files a bit ... actually I have added this:

```php
/**
     * Builds the service container.
     *
     * @return ContainerBuilder The compiled service container
     *
     * @throws \RuntimeException
     */
    protected function buildContainer()
    {
        foreach (['cache' => $this->getCacheDir(), 'build' => $this->warmupDir ?: $this->getBuildDir(), 'logs' => $this->getLogDir()] as $name => $dir) {
            if(is_dir($dir)){
                echo 'Directory Exist '. $dir .'<br>';
                echo get_current_user().'<br>';
                echo exec('whoami').'<br>';
                echo substr(sprintf('%o', fileperms($dir)), -4).'<br>';
                //chmod($dir, 755);
            }
            if (!is_dir($dir)) {
                if (false === @mkdir($dir, 0777, true) && !is_dir($dir)) {
                    throw new \RuntimeException(sprintf('Unable to create the "%s" directory (%s).', $name, $dir));
                }
            } elseif (!is_writable($dir)) {
                throw new \RuntimeException(sprintf('Unable to write in the "%s" directory (%s).', $name, $dir));
            }
        }

        $container = $this->getContainerBuilder();
        $container->addObjectResource($this);
        $this->prepareContainer($container);

        if (null !== $cont = $this->registerContainerConfiguration($this->getContainerLoader($container))) {
            trigger_deprecation('symfony/http-kernel', '5.3', 'Returning a ContainerBuilder from "%s::registerContainerConfiguration()" is deprecated.', get_debug_type($this));
            $container->merge($cont);
        }

        $container->addCompilerPass(new AddAnnotatedClassesToCachePass($this));

        return $container;
    }
```

this is the output:

![image](https://user-images.githubusercontent.com/1207306/144147261-c7da5d71-9d09-4743-a9c2-f64c828e88c5.png)

if I remove the comment ... and leave chmod the error stops happening ... if I comment it again it happens again ... this is something extremely strange ... someone can help me ... I am not supposed to alter directory permissions like this ...

**Comment by alessandro-podo** [comment]
your webserver is running as which User?
Which right does the Authenticated User have?

**Comment by wnunezc** [comment]
I have fixed it by deleting the cache dev folder by allowing symfony to recreate the folders again and doing it with the user running the symfony command console allows it to be created with the necessary permissions.

I honestly don't understand why it hasn't worked with post creation.


@alessandro-podo  I do not know this data precisely, let's say that I simply have a user in windows 10 and I assume that it is the same for everything ... Do you have any way to check it, any command for windows 10 that spits out the information in CMD or Shell ???

https://es.stackoverflow.com/a/499742/46896

**Reply by alessandro-podo** [reply]
use dump($_SERVER['USER']) than you see under which user it runs.

**Reply by wnunezc** [reply]
> use dump($_SERVER['USER']) than you see under which user it runs.

i have try this:

![image](https://user-images.githubusercontent.com/1207306/143960734-1d9f22aa-b799-4290-9e37-449ad73f01a1.png)

But i get this:

![image](https://user-images.githubusercontent.com/1207306/143960768-5310eb50-5c2b-4c13-8995-862b3f72289d.png)

**Reply by wnunezc** [reply]
PHP emviroment:

![image](https://user-images.githubusercontent.com/1207306/144066975-81c0e321-22f2-4fa8-8752-6e4298152646.png)

**Reply by alessandro-podo** [reply]
if you scroll down you find an section environment. which user is there?

**Reply by wnunezc** [reply]
> if you scroll down you find an section environment. which user is there?

not exist:

![image](https://user-images.githubusercontent.com/1207306/144077840-7e19dd41-907c-4cdc-a5e0-4b6533df0426.png)

**Reply by wnunezc** [reply]
> if you scroll down you find an section environment. which user is there?

i find this in aditional module section:

and the username figure like: wnune but this not exist in `$_SERVER`


![image](https://user-images.githubusercontent.com/1207306/144079967-9972e999-b1fd-419e-b274-485000de9fd2.png)

![image](https://user-images.githubusercontent.com/1207306/144080052-b3a3f5fe-7fa3-4d36-b251-d60755c1f85e.png)

**Reply by wnunezc** [reply]
> if you scroll down you find an section environment. which user is there?

but the user have the righ permison: 

![image](https://user-images.githubusercontent.com/1207306/144081123-9d843aa8-1236-44b2-bc05-1b1880de859b.png)

**Reply by wnunezc** [reply]
> if you scroll down you find an section environment. which user is there?

i have run this: `echo exec('whoami');` and show me:

> wnunez\wnunewnunez\wnune

**Reply by alessandro-podo** [reply]
sry than i have no answer :( 
my webserver run under Debian.
